curriculum/challenges/english/25-front-end-development/lab-lunch-picker-program/66db529d37ad966480ebb633.md

```diff
@@ -307,22 +307,57 @@ const temp = console.log;
 
 try {
   console.log = obj => tempArr.push(obj);
+  const regex = /Randomly selected lunch: \w+/  // Output format to compare results with
 
   // check that it correctly outputs the first item
   Math.random = () => 0;
   getRandomLunch(testLunches);
-  assert.strictEqual(tempArr[0], `Randomly selected lunch: ${testLunches[0]}`);
+  assert.strictEqual(regex.test(tempArr[0]), true);
 
-  // second item
+  // second item 
   Math.random = () => 0.4;
   getRandomLunch(testLunches);
-  assert.strictEqual(tempArr[1], `Randomly selected lunch: ${testLunches[1]}`);
+  assert.strictEqual(regex.test(tempArr[1]), true);
 
   // third item
   Math.random = () => 0.8;
   getRandomLunch(testLunches);
-  assert.strictEqual(tempArr[2], `Randomly selected lunch: ${testLunches[2]}`);
+  assert.strictEqual(regex.test(tempArr[2]), true);
+} finally {
+  // restore Math.random
+  Math.random = tempRandom;
+  console.log = temp;
+}
+```
+
+The function `getRandomLunch` should not modify the input array.
+
+```js
+const testLunches = ["Fish", "Fries", "Roast"];
+const tempRandom = Math.random;
```

Comment:
we do not need to do this if we are checking that the function does not change the array, this was to check that the random selection was implemented properly